enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,213 @@
+
+### Justification
+
+*If we don't make containerd as default runtime, we will be left out with Mirantis supported dockershim
+and will have to re-design WMCO to incorporate Mirantis dockershim.
+*From the business point of view subscribing to Mirantis for dockershim support is not an ideal situation as Mirantis
+ is our key competitor.
+*If we want to use CRI-O for Windows, the amount of time and engineering efforts involved making CRI-O to work for 
+Windows is huge. This doesn't do business justification due to the high cost and time. 
+*Containerd is widely adopted and supported by the open source community. 
+*Microsoft is a major contributor in Windows containerd development and moved to containerd as default runtime for their 
+kubernetes offerings. 
+*Most of the Windows-supported kubernetes orchestrators already moved to containerd.
+*Containerd has become the default runtime in linux/windows nodes.
+*Testing and adaptation will require less engineering effort compare to other solutions due to wider open source
+ community and enterprise support.
+
+### Design Details
+
+we plan to use containerd 1.6.0 to integrate into WMCO. Containerd will be installed as a first service before 
+kubelet installation as kubelet must require containerd to be running. We already bundled containerd as part of 
+WMCO package. As we can use ctr commandline to test containerd, crictl commandline will not be installed.
+Once containerd is installed, the rest of the service installation steps remain the same. Kubelet's dependency on Docker
+runtime will be removed and replaced by containerd. A separate folder will be created under C:\k\ to store containerd
+config and related files. In BYOH case, even after containerd becomes default runtime Docker will continue to run.
+User can choose to uninstall Docker or use it for non-kubernetes related requirements. As docker installation is
+handled by the user, it is up to the user to decide if Docker should continue to run or not.
+When it comes to MachineSet, we can choose a VM image which doesn't have Docker. Containerd will be the only runtime.
+Even if Docker gets installed at any time, it can co-exist with containerd. 
+steps to install containerd as service
+    *scp containerd/related executables into Windows VM
+    *copy the files in appropriate folder
+    *create containerd config file
+    *run below commands in PowerShell cmdlet
+    *run extra command for performance `Add-MpPreference -ExclusionProcess "$Env:containerd.exe"`
+    *register containerd as service
+    *start containerd as a Windows service
+
+## Network changes
+Current CNI/IPAM will be used for containerd and no change in HNS-Network and HNS-Endpoint creation
+steps. The config file which will be used by containerd will point to same CNI/IPAM executables.
+
+## WMCO feature flag
+Containerd feature flag will be used to enable this feature. This will be enabled by default in
```

Comment:
Isn't this the runtime feature flag? In it's absence it will default to Docker in 4.10. In the 4.10 community operator it will be specified i.e. `--runtime containerd`. In 4.11 it will be removed. That is what needs to be explained here. 

I suggest calling this a WMCO CLI option.